canonical-web-and-design/ubuntu.com
# TCO calculator rebuild

## Done

- Remove logic needed for old tco-calculator
- Add react component for new calculator, all calculations are based on this [spreadsheet](https://docs.google.com/spreadsheets/d/1KrR3s21K2uyjYtwHHnPHiN91byK5SXKxU-NXdCA8MbY/edit#gid=1544469160)
- Update secondary nav and path to `openstack/pricing-calculator`, and add a redirect from `/openstack/tco-calculator`
- Build new page including new calculator
- Integrate enzyme testing framework and add tests
- Add new contact us form for the estimates - for MVP there is no option to add another instance


## QA

- View the page at: https://ubuntu-com-9974.demos.haus/openstack/pricing-calculator
- First check the copy and design based on this [copy doc](https://docs.google.com/document/d/1n3ExZwFVlw5LyAp7O3CgPthVgx8yRX5-Wg_gBQLf73c/edit#) and this [design](https://app.zeplin.io/project/60d33b8cbfc46b12b0b4d3cb/screen/60d33babdf4648121efe1e13)
- Go to the [spreadsheet](https://docs.google.com/spreadsheets/d/1KrR3s21K2uyjYtwHHnPHiN91byK5SXKxU-NXdCA8MbY/edit#gid=1544469160), and change the inputs on the `Calculator` tab. (e.g. Number of instances, vCPUs etc)
- Go back to https://ubuntu-com-9974.demos.haus/openstack/pricing-calculator and change the values on the inputs on the calculator to be exactly the same as the spreadsheet. Check the outputs are the same. 

- NOTE: `Get the free report` and `Learn more about TCO calculators` links to a copy doc of an engage page which doesn't exist yet. 

## Issue / Card

Fixes [#4197](https://github.com/canonical-web-and-design/web-squad/issues/4197)

## Screenshots

![Screenshot 2021-06-30 at 11 12 44](https://user-images.githubusercontent.com/58959073/123943498-1ffd1900-d994-11eb-8444-645d059c377f.png)
